cloud/endagaweb/tests/test_denomination.py

```diff
@@ -0,0 +1,107 @@
+"""Tests for models.Users.
+
+Copyright (c) 2016-present, Facebook, Inc.
+All rights reserved.
+
+This source code is licensed under the BSD-style license found in the
+LICENSE file in the root directory of this source tree. An additional grant
+of patent rights can be found in the PATENTS file in the same directory.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from datetime import datetime
+from random import randrange
+import uuid
+from django import test
+import json
+
+import pytz
+
+from django.test import TestCase
+
+from ccm.common import crdt
+from endagaweb import models
+
+
+class TestBase(TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        cls.username = 'y'
+        cls.password = 'pw'
+        cls.user = models.User(username=cls.username, email='y@l.com')
+        cls.user.set_password(cls.password)
+        cls.user.save()
+        cls.user_profile = models.UserProfile.objects.get(user=cls.user)
+
+        cls.uuid = "59216199-d664-4b7a-a2db-6f26e9a5d208"
+
+        # Create a test client.
+        cls.client = test.Client()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.user.delete()
+        cls.user_profile.delete()
+
+    def tearDown(self):
+        self.logout()
+
+    def login(self):
+        """Log the client in."""
+        data = {
+            'email': self.username,
+            'password': self.password,
+        }
+        self.client.post('/auth/', data)
+
+    def logout(self):
+        """Log the client out."""
+        self.client.get('/logout')
+
+
+class DenominationUITest(TestBase):
+    """Testing that we can add User in the UI."""
+
+    def test_add_denominaton(self):
+        self.logout()
+        response = self.client.get('/dashboard/network/denominations')
+        # Anonymous User can not see this page so returning  permission denied.
+        self.assertEqual(302, response.status_code)
```

Comment:
302 doesn't seem like the correct return code here, unless it's an artifact of authentication failure being translated into redirection back to the login page. That doesn't seem like the right behaviour an API call though.